test/kinetics/kineticsFromYaml.cpp

```diff
@@ -162,15 +225,16 @@ TEST(Reaction, ChebyshevFromYaml)
 
     auto R = newReaction(rxn, *(sol->kinetics()));
     EXPECT_EQ(R->reactants.size(), (size_t) 1);
-    auto CR = dynamic_cast<ChebyshevReaction&>(*R);
+    auto CR = dynamic_cast<ChebyshevReaction3&>(*R);
+    const auto& rate = std::dynamic_pointer_cast<ChebyshevRate3>(CR.rate());
```

Comment:
Another reason for my suggestion of having the `rate()` member function be part of the `Reaction` class is it would make the first `dynamic_cast` here unnecessary in cases like this where the only thing you're after is the `ReactionRate` object and not any behavior defined only for the `ChebyshevRate3` object (of which there is actually none).